test/unit/stores/test_image.py

```diff
@@ -85,6 +85,33 @@ def test_upload(self):
         self.check_upload_success()
 
 
+class DeleteDSL(ImageStoreDSL):
+    """Base class for `delete` tests."""
+
+    def call_delete(self) -> None:
+        """Calls the `ImageStore` `delete` method."""
+
+        self.image_store.delete("object_key")
```

Comment:
Not necessary but for consistency I would prefer taking in the image_id, store it in a `_delete_system_id: str` and then assert using that like we do for repo tests to be consistent.